### PR TITLE
CI: Add actions for npm release

### PR DESCRIPTION
**Pull request by imWildCat**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        with:
+          node-version: 14
+      - name: match-tag-to-package-version
+        uses: geritol/match-tag-to-package-version@0.0.2
+        env:
+          TAG_PREFIX: refs/tags/v
+      - run: npm install
+      - run: npm version ${{ github.event.release.tag_name }}
+      - run: npm run build
+      - run: npm config set '//registry.npmjs.org/:_authToken' "${{ secrets.NPM_TOKEN }}"
```
Comment (imWildCat) [inline .github/workflows/release.yml:27]: I've created this token.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+name: Release Pipeline
+
+on:
+  release:
+    types: [ published ]
```
Comment (zhuorantan) [inline .github/workflows/release.yml:5]: Do we also publish to NPM for prereleases? If so I think we need to add a `prerelease` type (not sure if it's the exact name).

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+jobs:
+  npm-relase:
+    name: 'Release to npm'
+    if: ${{ startsWith(github.ref, 'refs/tags/') }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.release.target_commitish }}
```
Comment (zhuorantan) [inline .github/workflows/release.yml:15]: Do we need specify this? I thought this was the default.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+name: Release Pipeline
+
+on:
+  release:
+    types: [ published ]
+
+jobs:
+  npm-relase:
+    name: 'Release to npm'
+    if: ${{ startsWith(github.ref, 'refs/tags/') }}
```
Comment (zhuorantan) [inline .github/workflows/release.yml:10]: Also, since we are already using release trigger, this is not need right?

Comment (imWildCat) [reply .github/workflows/release.yml]: Removed.

File: .github/workflows/release.yml (hunk 2 above)
Comment (imWildCat) [reply .github/workflows/release.yml]: Yeah.

Thanks for the reminder!

**Review by zhuorantan — APPROVED** [review]

**Review by zhuorantan — APPROVED** [review]

**Pull request merged**